src/Avalonia.Base/Media/RenderOptions.cs

```diff
@@ -1,36 +1,181 @@
+        /// <returns>The control's left coordinate.</returns>
+        public static TextRenderingMode GetTextRenderingMode(Visual element)
+        {
+            return element.RenderOptions.TextRenderingMode;
+        }
+
+        /// <summary>
+        /// Sets the value of the TextRenderingMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <param name="value">The left value.</param>
+        public static void SetTextRenderingMode(Visual element, TextRenderingMode value)
+        {
+            element.RenderOptions = element.RenderOptions.WithTextRenderingMode(value);
+        }
+
+        public RenderOptions WithBitmapInterpolationMode(BitmapInterpolationMode bitmapInterpolationMode)
+        {
+            return new RenderOptions
+            {
+                BitmapInterpolationMode = bitmapInterpolationMode,
+                EdgeMode = EdgeMode,
+                TextRenderingMode= TextRenderingMode
+            };
+        }
+
+        public RenderOptions WithBitmapBlendingMode(BitmapBlendingMode bitmapBlendingMode)
+        {
+            return new RenderOptions
+            {
+                BitmapInterpolationMode = BitmapInterpolationMode,
+                EdgeMode = EdgeMode,
+                TextRenderingMode= TextRenderingMode
+            };
+        }
+
+        public RenderOptions WithEdgeMode(EdgeMode edgeMode)
+        {
+            return new RenderOptions
+            {
+                BitmapInterpolationMode = BitmapInterpolationMode,
+                EdgeMode = edgeMode,
+                TextRenderingMode = TextRenderingMode
+            };
+        }
+
+        public RenderOptions WithTextRenderingMode(TextRenderingMode textRenderingMode)
+        {
+            return new RenderOptions 
+            {
+                BitmapInterpolationMode = BitmapInterpolationMode,
+                EdgeMode = EdgeMode,
+                TextRenderingMode = textRenderingMode 
+            };
+        }
+
+        public RenderOptions MergeWith(RenderOptions other)
+        {
+            var bitmapInterpolationMode = BitmapInterpolationMode;
+
+            if(bitmapInterpolationMode == BitmapInterpolationMode.Default)
+            {
+                bitmapInterpolationMode = other.BitmapInterpolationMode;
+            }
+
+
+            var edgeMode = EdgeMode;
+
+            if(edgeMode == EdgeMode.Antialias)
+            {
+                edgeMode = other.EdgeMode;
+            }
+
+            var textRenderingMode = TextRenderingMode;
+
+            if(textRenderingMode == TextRenderingMode.SubpixelAntialias)
+            {
+                textRenderingMode= other.TextRenderingMode;
+            }
+
+            return new RenderOptions
+            {
+                BitmapInterpolationMode = bitmapInterpolationMode,
+                EdgeMode = edgeMode,
+                TextRenderingMode = textRenderingMode
+            };
+        }
```

Comment:
Small nit. This struct can be made a record. Which also would provide compiler support of "with" keyword. And some other methods can be removed as well.